### PR TITLE
Update dependencies and prepare release

### LINKED ISSUE #1360: [Bug] `bee-pow 0.2.0` compilation fails with `iota-crypto 0.9.2`
## Bug description

The release of the `iota-crypto 0.9.2` patch to crates.io today (2022-04-25) broke compilation for the `bee-pow 0.2.0` crate, imported by `iota-client`. This is causing compilation errors for the Identity team.

The error log suggests that this is due to the version mismatch of the `bee-ternary` crate:

- `iota-crypto 0.9.2` uses `bee-ternary 0.6.0` [(Cargo.toml)](https://github.com/iotaledger/crypto.rs/blob/864f3a03f1efa362b325663627060d78dae881b4/Cargo.toml#L69)
- `bee-pow 0.2.0` uses `bee-ternary 0.5.2` [(Cargo.toml)](https://github.com/iotaledger/bee/blob/b98bd114e763a0cebe47ac4b8055873e8009e8e6/bee-pow/Cargo.toml#L14)

Note that `bee-pow` is not pinned to `iota-crypto 0.9.1` (which used `bee-ternary 0.5.2`), so it pulls in the latest `0.9.2` version causing this issue.
```
[dependencies]
bee-ternary = { version = "0.5.2", default-features = false }
iota-crypto = { version = "0.9.1", default-features = false, features = [ "blake2b", "digest", "curl-p" ] }
```

The pertinent error is as follows, note the helpful message: "perhaps two different versions of crate `bee_ternary` are being used?"

```rust
error[E0308]: mismatched types
  --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-pow-0.2.0\src\score.rs:52:9
   |
33 |     pub fn hash(&mut self, bytes: &[u8]) -> TritBuf<T1B1Buf> {
   |                                             ---------------- expected `TritBuf` because of return type
...
52 |         self.curl.digest(self.pow_input.as_slice())
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `TritBuf`, found struct `bee_ternary::TritBuf`
   |
   = note: expected struct `TritBuf`
              found struct `bee_ternary::TritBuf`
   = note: perhaps two different versions of crate `bee_ternary` are being used?
```
(full error log at end of issue)

We currently work around this problem by pinning to `iota-crypto 0.9.1` in our project locally, but would appreciate an update to `bee-pow` so it uses `bee-ternary 0.6.0` and `iota-crypto 0.9.2` (or pins the current dependency versions) and release a 0.2.1 patch, if at all possible. Other avenues are asking `iota-crypto` to yank `0.9.2` and release a minor update like `0.10.0` instead, that `bee-pow 0.2.0` would not pull in automatically.

## Rust version

Which version of Rust are you running?

- Rust version: 
```
❯ cargo version
cargo 1.60.0 (d1fd9fe2c 2022-03-01)  
❯ rustc --version
rustc 1.60.0 (7737e0b5c 2022-04-04) 
```

## Bee version

Which version of Bee are you using?

- Bee version (version number, commit, or branch): `bee-pow 0.2.0` published to crates.io (imported transitively via `bee-message 0.1,6` and `iota-client 1.1.1`).

## Steps To reproduce the bug

Minimum reproduction is as follows:

1. Create a new Rust project which imports the latest published version of `bee-pow`.
```
[package]
name = "untitled"
version = "0.1.0"
edition = "2021"

[dependencies]
bee-pow = "0.2.0"
```
2. Attempt to compile with `cargo check`.

## Expected behaviour

`bee-pow` compiles with no error.

## Actual behaviour

`cargo` fails and throws the following error log.

## Errors

```rust
❯ cargo check
   Compiling autocfg v1.1.0      
   Compiling typenum v1.15.0     
   Compiling version_check v0.9.4
   Compiling proc-macro2 v1.0.37
   Compiling unicode-xid v0.2.2
   Compiling syn v1.0.91
    Checking subtle v2.4.1
    Checking spin v0.5.2
    Checking opaque-debug v0.3.0
    Checking lazy_static v1.4.0
   Compiling generic-array v0.14.5
   Compiling num-traits v0.2.14
   Compiling bee-ternary v0.6.0
   Compiling bee-ternary v0.5.2
   Compiling quote v1.0.18
    Checking crypto-mac v0.8.0
    Checking digest v0.9.0
    Checking blake2 v0.9.2
    Checking iota-crypto v0.9.2
   Compiling thiserror-impl v1.0.30
    Checking thiserror v1.0.30
    Checking bee-pow v0.2.0
error[E0599]: the function or associated item `new` exists for struct `CurlPBatchHasher<T1B1Buf>`, but its trait bounds were not satisfied
   --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-pow-0.2.0\src\providers\miner.rs:116:55
    |
116 |         let mut hasher = CurlPBatchHasher::<T1B1Buf>::new(HASH_LENGTH);
    |                                                       ^^^ function or associated item cannot be called on `CurlPBatchHasher<T1B1Buf>` due to unsatisfied trait bounds
    |
   ::: C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-ternary-0.5.2\src\t1b1.rs:102:1
    |
102 | pub struct T1B1Buf<T: Trit = Btrit>(Vec<T>);
    | -------------------------------------------- doesn't satisfy `T1B1Buf: bee_ternary::raw::RawEncodingBuf`
    |
   ::: C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\iota-crypto-0.9.2\src\hashes\ternary\curl_p\batched\mod.rs:31:1
    |
31  | pub struct CurlPBatchHasher<B: RawEncodingBuf> {
    | ----------------------------------------------
    | |
    | doesn't satisfy `CurlPBatchHasher<T1B1Buf>: Clone`
    | doesn't satisfy `CurlPBatchHasher<T1B1Buf>: Default`
    | doesn't satisfy `CurlPBatchHasher<T1B1Buf>: Digest`
    | doesn't satisfy `CurlPBatchHasher<T1B1Buf>: digest::Reset`
    | doesn't satisfy `CurlPBatchHasher<T1B1Buf>: digest::Update`
    | doesn't satisfy `_: digest::fixed::FixedOutput`
    |
    = note: the following trait bounds were not satisfied:
            `T1B1Buf: bee_ternary::raw::RawEncodingBuf`
            `CurlPBatchHasher<T1B1Buf>: digest::Update`
            which is required by `CurlPBatchHasher<T1B1Buf>: Digest`
            `CurlPBatchHasher<T1B1Buf>: digest::fixed::FixedOutput`
            which is required by `CurlPBatchHasher<T1B1Buf>: Digest`
            `CurlPBatchHasher<T1B1Buf>: digest::Reset`
            which is required by `CurlPBatchHasher<T1B1Buf>: Digest`
            `CurlPBatchHasher<T1B1Buf>: Clone`
            which is required by `CurlPBatchHasher<T1B1Buf>: Digest`
            `CurlPBatchHasher<T1B1Buf>: Default`
            which is required by `CurlPBatchHasher<T1B1Buf>: Digest`
            `&CurlPBatchHasher<T1B1Buf>: digest::Update`
            which is required by `&CurlPBatchHasher<T1B1Buf>: Digest`
            `&CurlPBatchHasher<T1B1Buf>: digest::fixed::FixedOutput`
            which is required by `&CurlPBatchHasher<T1B1Buf>: Digest`
            `&CurlPBatchHasher<T1B1Buf>: digest::Reset`
            which is required by `&CurlPBatchHasher<T1B1Buf>: Digest`
            `&CurlPBatchHasher<T1B1Buf>: Default`
            which is required by `&CurlPBatchHasher<T1B1Buf>: Digest`
            `&mut CurlPBatchHasher<T1B1Buf>: digest::Update`
            which is required by `&mut CurlPBatchHasher<T1B1Buf>: Digest`
            `&mut CurlPBatchHasher<T1B1Buf>: digest::fixed::FixedOutput`
            which is required by `&mut CurlPBatchHasher<T1B1Buf>: Digest`
            `&mut CurlPBatchHasher<T1B1Buf>: digest::Reset`
            which is required by `&mut CurlPBatchHasher<T1B1Buf>: Digest`
            `&mut CurlPBatchHasher<T1B1Buf>: Clone`
            which is required by `&mut CurlPBatchHasher<T1B1Buf>: Digest`
            `&mut CurlPBatchHasher<T1B1Buf>: Default`
            which is required by `&mut CurlPBatchHasher<T1B1Buf>: Digest`

error[E0277]: the trait bound `T1B1Buf: bee_ternary::raw::RawEncodingBuf` is not satisfied
   --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-pow-0.2.0\src\providers\miner.rs:116:26
    |
116 |         let mut hasher = CurlPBatchHasher::<T1B1Buf>::new(HASH_LENGTH);
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `bee_ternary::raw::RawEncodingBuf` is not implemented for `T1B1Buf`
    |
note: required by a bound in `CurlPBatchHasher`
   --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\iota-crypto-0.9.2\src\hashes\ternary\curl_p\batched\mod.rs:31:32
    |
31  | pub struct CurlPBatchHasher<B: RawEncodingBuf> {
    |                                ^^^^^^^^^^^^^^ required by this bound in `CurlPBatchHasher`

error[E0308]: mismatched types
  --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-pow-0.2.0\src\score.rs:52:26
   |
52 |         self.curl.digest(self.pow_input.as_slice())
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `bee_ternary::Trits`, found struct `Trits`
   |
   = note: expected reference `&bee_ternary::Trits`
              found reference `&Trits`
   = note: perhaps two different versions of crate `bee_ternary` are being used?

error[E0308]: mismatched types
  --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-pow-0.2.0\src\score.rs:52:9
   |
33 |     pub fn hash(&mut self, bytes: &[u8]) -> TritBuf<T1B1Buf> {
   |                                             ---------------- expected `TritBuf` because of return type
...
52 |         self.curl.digest(self.pow_input.as_slice())
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `TritBuf`, found struct `bee_ternary::TritBuf`
   |
   = note: expected struct `TritBuf`
              found struct `bee_ternary::TritBuf`
   = note: perhaps two different versions of crate `bee_ternary` are being used?

Some errors have detailed explanations: E0277, E0308, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `bee-pow` due to 4 previous errors                                                                                                                                     00:11:05  
C:\Work\untitled main ❯ cargo check
    Checking bee-pow v0.2.0
error[E0599]: the function or associated item `new` exists for struct `CurlPBatchHasher<T1B1Buf>`, but its trait bounds were not satisfied
   --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-pow-0.2.0\src\providers\miner.rs:116:55
    |
116 |         let mut hasher = CurlPBatchHasher::<T1B1Buf>::new(HASH_LENGTH);
    |                                                       ^^^ function or associated item cannot be called on `CurlPBatchHasher<T1B1Buf>` due to unsatisfied trait bounds
    |
   ::: C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-ternary-0.5.2\src\t1b1.rs:102:1
    |
102 | pub struct T1B1Buf<T: Trit = Btrit>(Vec<T>);
    | -------------------------------------------- doesn't satisfy `T1B1Buf: bee_ternary::raw::RawEncodingBuf`
    |
   ::: C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\iota-crypto-0.9.2\src\hashes\ternary\curl_p\batched\mod.rs:31:1
    |
31  | pub struct CurlPBatchHasher<B: RawEncodingBuf> {
    | ----------------------------------------------
    | |
    | doesn't satisfy `CurlPBatchHasher<T1B1Buf>: Clone`
    | doesn't satisfy `CurlPBatchHasher<T1B1Buf>: Default`
    | doesn't satisfy `CurlPBatchHasher<T1B1Buf>: Digest`
    | doesn't satisfy `CurlPBatchHasher<T1B1Buf>: digest::Reset`
    | doesn't satisfy `CurlPBatchHasher<T1B1Buf>: digest::Update`
    | doesn't satisfy `_: digest::fixed::FixedOutput`
    |
    = note: the following trait bounds were not satisfied:
            `T1B1Buf: bee_ternary::raw::RawEncodingBuf`
            `CurlPBatchHasher<T1B1Buf>: digest::Update`
            which is required by `CurlPBatchHasher<T1B1Buf>: Digest`
            `CurlPBatchHasher<T1B1Buf>: digest::fixed::FixedOutput`
            which is required by `CurlPBatchHasher<T1B1Buf>: Digest`
            `CurlPBatchHasher<T1B1Buf>: digest::Reset`
            which is required by `CurlPBatchHasher<T1B1Buf>: Digest`
            `CurlPBatchHasher<T1B1Buf>: Clone`
            which is required by `CurlPBatchHasher<T1B1Buf>: Digest`
            `CurlPBatchHasher<T1B1Buf>: Default`
            which is required by `CurlPBatchHasher<T1B1Buf>: Digest`
            `&CurlPBatchHasher<T1B1Buf>: digest::Update`
            which is required by `&CurlPBatchHasher<T1B1Buf>: Digest`
            `&CurlPBatchHasher<T1B1Buf>: digest::fixed::FixedOutput`
            which is required by `&CurlPBatchHasher<T1B1Buf>: Digest`
            `&CurlPBatchHasher<T1B1Buf>: digest::Reset`
            which is required by `&CurlPBatchHasher<T1B1Buf>: Digest`
            `&CurlPBatchHasher<T1B1Buf>: Default`
            which is required by `&CurlPBatchHasher<T1B1Buf>: Digest`
            `&mut CurlPBatchHasher<T1B1Buf>: digest::Update`
            which is required by `&mut CurlPBatchHasher<T1B1Buf>: Digest`
            `&mut CurlPBatchHasher<T1B1Buf>: digest::fixed::FixedOutput`
            which is required by `&mut CurlPBatchHasher<T1B1Buf>: Digest`
            `&mut CurlPBatchHasher<T1B1Buf>: digest::Reset`
            which is required by `&mut CurlPBatchHasher<T1B1Buf>: Digest`
            `&mut CurlPBatchHasher<T1B1Buf>: Clone`
            which is required by `&mut CurlPBatchHasher<T1B1Buf>: Digest`
            `&mut CurlPBatchHasher<T1B1Buf>: Default`
            which is required by `&mut CurlPBatchHasher<T1B1Buf>: Digest`

error[E0277]: the trait bound `T1B1Buf: bee_ternary::raw::RawEncodingBuf` is not satisfied
   --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-pow-0.2.0\src\providers\miner.rs:116:26
    |
116 |         let mut hasher = CurlPBatchHasher::<T1B1Buf>::new(HASH_LENGTH);
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `bee_ternary::raw::RawEncodingBuf` is not implemented for `T1B1Buf`
    |
note: required by a bound in `CurlPBatchHasher`
   --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\iota-crypto-0.9.2\src\hashes\ternary\curl_p\batched\mod.rs:31:32
    |
31  | pub struct CurlPBatchHasher<B: RawEncodingBuf> {
    |                                ^^^^^^^^^^^^^^ required by this bound in `CurlPBatchHasher`

error[E0308]: mismatched types
  --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-pow-0.2.0\src\score.rs:52:26
   |
52 |         self.curl.digest(self.pow_input.as_slice())
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `bee_ternary::Trits`, found struct `Trits`
   |
   = note: expected reference `&bee_ternary::Trits`
              found reference `&Trits`
   = note: perhaps two different versions of crate `bee_ternary` are being used?

error[E0308]: mismatched types
  --> C:\Users\cycra\.cargo\registry\src\github.com-1ecc6299db9ec823\bee-pow-0.2.0\src\score.rs:52:9
   |
33 |     pub fn hash(&mut self, bytes: &[u8]) -> TritBuf<T1B1Buf> {
   |                                             ---------------- expected `TritBuf` because of return type
...
52 |         self.curl.digest(self.pow_input.as_slice())
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `TritBuf`, found struct `bee_ternary::TritBuf`
   |
   = note: expected struct `TritBuf`
              found struct `bee_ternary::TritBuf`
   = note: perhaps two different versions of crate `bee_ternary` are being used?

Some errors have detailed explanations: E0277, E0308, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `bee-pow` due to 4 previous errors                         
```

### PR DESCRIPTION
**Pull request by grtlr**
# Description of change

The following packages will be released as crates to crates.io. 

Closes #1360.

* `bee-pow`
* `bee-message`
* `bee-ledger`

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-ternary/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+
```
Comment (thibault-martinez) [inline bee-ternary/CHANGELOG.md:27]:
```suggestion
```

File: bee-node/bee-plugin/bee-plugin-dashboard/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "bee-plugin-dashboard"
-version = "0.1.0"
+version = "0.2.0"
```
Comment (thibault-martinez) [inline bee-node/bee-plugin/bee-plugin-dashboard/Cargo.toml:3]: Just curious, why was this one bumped but not the MPS plugin ?

Comment (grtlr) [reply bee-node/bee-plugin/bee-plugin-dashboard/Cargo.toml:3]: Yeah, I don't think we need the bump here. Should I undo it?

Comment (thibault-martinez) [reply bee-node/bee-plugin/bee-plugin-dashboard/Cargo.toml:3]: What is your reasoning for bumping or not here ?

File: bee-ternary/Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "bee-ternary"
-version = "0.5.2"
+version = "0.6.0"
```
Comment (thibault-martinez) [inline bee-ternary/Cargo.toml:3]: Sorry being annoying but there is no changelog for this 😬

File: bee-storage/bee-storage-test/CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Update dependencies;
+
```
Comment (thibault-martinez) [inline bee-storage/bee-storage-test/CHANGELOG.md]: Remove

File: bee-storage/bee-storage-memory/CHANGELOG.md [hunk 5]
```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Update dependencies;
+
```
Comment (thibault-martinez) [inline bee-storage/bee-storage-memory/CHANGELOG.md]: Remove

File: bee-node/bee-plugin/bee-plugin-mps/CHANGELOG.md [hunk 6]
```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- Updated dependencies;
+
```
Comment (thibault-martinez) [inline bee-node/bee-plugin/bee-plugin-mps/CHANGELOG.md]: Remove

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**